build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryDependenciesClient.java

```diff
@@ -194,6 +199,27 @@ public boolean isArtifactoryOSS() throws IOException {
         return !httpClient.getVersion().hasAddons();
     }
 
+    public boolean isRemoteRepo(String repo) throws IOException {
+        PreemptiveHttpClient client = httpClient.getHttpClient();
+        String url = artifactoryUrl + "/api/repositories/" + repo;
+        HttpGet httpGet = new HttpGet(url);
+        InputStream responseStream = getResponseStream(client.execute(httpGet), "Failed to retrieve repository configuration '" + repo + "'");
+        RepositoryResult repoDetails = readJsonResponse(responseStream,
+                new TypeReference<RepositoryResult>() {
+                },
+                true);
+        return "remote".equals(repoDetails.getRclass());
+    }
+
+    public void downloadMarkerLayer(String repo, String imageName, String ImageMarkerName) throws IOException {
+        String url = artifactoryUrl + "/api/docker/" + repo + "/v2/" + imageName + "/blobs/" + toNoneMarkerLayer(ImageMarkerName);
+        this.getArtifactMetadata(url);
+    }
+
+    private String toNoneMarkerLayer(String markerLayerName) {
+        return markerLayerName.substring(0, markerLayerName.length() - ".marker".length());
+    }
```

Comment:
These methods are not related to the Dependencies client. Please move them to DockerUtils.